lightning/src/ln/peer_handler.rs

```diff
@@ -1224,12 +1224,30 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 						if let Some((announce, update_a_option, update_b_option)) =
 							self.message_handler.route_handler.get_next_channel_announcement(c)
 						{
-							self.enqueue_message(peer, &announce);
+							let (min, max) = peer.sent_gossip_timestamp_filter.as_ref().map_or((0, u32::MAX), |f| {
+								(f.first_timestamp, f.first_timestamp.saturating_add(f.timestamp_range))
+							});
+
+							// we only wish to enqueue the announcement if at least one directional
+							// update is going to accompany it
+							let mut has_enqueued_announcement = false;
+
 							if let Some(update_a) = update_a_option {
-								self.enqueue_message(peer, &update_a);
+								if update_a.contents.timestamp >= min && update_a.contents.timestamp <= max {
+									if !has_enqueued_announcement {
```

Comment:
Because `has_enqueued_announcement` is initialized to `false` and never modified before reaching this line. The variable is actually only used to prevent enqueuing the announcement more than once. The reason give in the comment above is really covered by the `let if let Some(update_a|b)` checks.